libs/designsystem/src/lib/components/avatar/avatar.component.ts

```diff
@@ -21,7 +21,7 @@ export class AvatarComponent {
   @Input() shadow: boolean;
   @Input() text: string;
   @Input() overlay: boolean;
-  @Input() size: AvatarSize = AvatarSize.SM;
+  @Input() size: AvatarSize | keyof AvatarSize = AvatarSize.SM;
```

Comment:
Unfortunately `keyof AvatarSize` returns all of the keys for a string object. 

I've created an example here: [link to typescript playground](https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBMAdgVwLZwIIDcCGMdQDKAlgF7BwDeAUHHABqFwC8cA5CAM5sA0tchALIt2nVL36CAIiLaoAJhLoAZAOKyANgHMJAX2rUYATzAUAKsE4wzJiq2x4CJcnAA+cANbAjEAGaZcfCIyYABuIA).

Notice the type of `TestType`
![Screenshot 2022-03-08 at 14 52 58](https://user-images.githubusercontent.com/17269698/157251757-70fddc0a-ed84-48eb-9ca2-ea71b641d33b.png)

I think we need something like `valueof AvatarSize`. 

This comment goes for all of the places where `keyof` has been used.

Reply:
It is only the case where the case in the enum maps to a string. I already fixed it for `ButtonSize`, but didn't see it for `AvatarSize`. I found 2 others.